codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostRequestEncoder.java

```diff
@@ -206,9 +215,8 @@ public HttpPostRequestEncoder(
             throw new NullPointerException("charset");
         }
         HttpMethod method = request.method();
-        if (!(method.equals(HttpMethod.POST) || method.equals(HttpMethod.PUT)
-                || method.equals(HttpMethod.PATCH) || method.equals(HttpMethod.OPTIONS))) {
-            throw new ErrorDataEncoderException("Cannot create a Encoder if not a POST");
+        if (method.equals(HttpMethod.TRACE)) {
+            throw new ErrorDataEncoderException("Cannot create a Encoder if request is a TRACE");
```

Comment:
any reason why this is not `IllegalArgumentException`? We are not necessarily encoding data here.